src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1118,7 +1118,12 @@ public new MetadataReference GetMetadataReference(IAssemblySymbol assemblySymbol
 
         private protected override MetadataReference CommonGetMetadataReference(IAssemblySymbol assemblySymbol)
         {
-            return GetMetadataReference(assemblySymbol.EnsureCSharpSymbolOrNull(nameof(assemblySymbol)));
+            if (assemblySymbol is Symbols.PublicModel.Symbol { UnderlyingSymbol: AssemblySymbol underlyingSymbol })
+            {
+                return GetMetadataReference(underlyingSymbol);
+            }
+
+            return null;
```

Comment:
>return null; [](start = 12, length = 12)

Does this maintain the original behavior when the input is null?

Reply:
Yes, will add the test.

---
In reply to: [364426266](https://github.com/dotnet/roslyn/pull/40794#discussion_r364426266) [](ancestors = 364426266)